test/test_misc.py

```diff
@@ -2052,6 +2052,21 @@ def fn(x, m):
             res = fn(x, m)
         self.assertEqual(ref, res)
 
+    def test_tensor_dot_grad_no_graph_break(self):
+        def fn(a, b):
+            y = 3 * a**3 - b**2
+            y.backward(gradient=torch.tensor([1.0, 1.0]))
```

Comment:
what was failing the is_tensor check here previously?

Reply:
Actually it's not failure, previously it wraps tensor ```a``` and tensor ```b``` as ```UserDefinedObjectVariable```(the fallback variable), so cause more graph break when accessing ```.grad```.